### DISCUSSION TITLE
How do you define a function?

### DISCUSSION THREAD

**Discussion by leshow** [vectordotdev/vrl#1035] [Q&A] [answered]
I'm trying to figure out how to define a function inside VRL. I assumed because you can pass a function with `|foo| {}` that this would work on assignment as well, but it doesn't appear to be valid. I haven't been able to find it in the docs, what am I missing?
looking for something like
```
my_fn = |val| {
...
}
```

so that I can call it later with `my_fn(foo)`

**Comment by jszwedko** [comment, marked as answer]
Unfortunately this isn't possible yet, see https://github.com/vectordotdev/vrl/issues/47
